File: src/main/java/mods/eln/sim/electrical/process/DiodeProcess.java
package mods.eln.sim.electrical.process;

import mods.eln.sim.IProcess;
import mods.eln.sim.electrical.mna.component.ResistorSwitch;

public class DiodeProcess implements IProcess {

    ResistorSwitch resistor;

    public DiodeProcess(ResistorSwitch resistor) {
        this.resistor = resistor;
    }

    @Override
    public void process(double time) {
        resistor.setState(resistor.getU() > 0);
    }
}

File: src/main/java/mods/eln/sim/DiodeProcess.kt
package mods.eln.sim

import mods.eln.sim.mna.component.ResistorSwitch

class DiodeProcess(internal var resistor: ResistorSwitch) : IProcess {

    override fun process(time: Double) {
        resistor.setState(resistor.getVoltage() > 0)
    }
}
